### ISSUE TITLE
@types/next-redux-wrapper: return value of withRedux "does not have any construct or call signatures"

### ISSUE THREAD

**Issue by bensaufley** [DefinitelyTyped/DefinitelyTyped#24293]
- [X] I tried using the `@types/next-redux-wrapper` package and had problems.
- [X] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [X] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [X] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @stevegeek, @andy-ms 

Looking at the [definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/next-redux-wrapper/index.d.ts), `withRedux` appears to return a `NextReduxWrappedComponent` which is a type composed of `React.Component & NextPageComponentMethods`. But treating the export as a React Component presents the following TypeScript error: `JSX element type 'MyComponent' does not have any construct or call signatures.` Diving into my `node_modules` directory I was able to update the definition of `NextReduxWrappedComponent` to this:

```ts
type NextReduxWrappedComponent<P, S> = React.Component<P, S> & NextPageComponentMethods;
```

~~And the error went away. But I'm still new enough to TypeScript that I have no idea if this is a _good_ solution or if I'm solving the problem the wrong way.~~ EDIT: see comments below; this definitely isn't the fix.

I've managed to make [an example repo](https://github.com/bensaufley/example-next-redux-wrapper-problem). You can either run `npm run tslint` which uses the (deprecated) `--type-check` or you should be able to open it in something like Visual Studio Code to see the error from `[ts]` like so:

![image](https://user-images.githubusercontent.com/331287/37432003-4d922366-27ad-11e8-9c45-7537a32ffd2b.png)

**Comment by BrunoScheufler** [comment]
@bensaufley same here, did you find any solution until now?

**Comment by bensaufley** [comment]
@BrunoScheufler nothing has changed on my end, so yeah we're still encountering this. It doesn't prevent test or build, but it does give us false positives when linting. And there's these squigglies.

**Comment by bensaufley** [comment]
I thought maybe I'd throw together a PR with the `<P, S>` additions but as I feared that seems to create upstream issues: references to `nextReduxWrapper.NextReduxWrappedComponent` on L28 and L35 then say `Generic type `NextReduxWrappedComponent` requires 2 type argument(s).` Adding some type arguments there means I'm back at the original problem. Can't find a place to pass down types to the Generic that makes the issue go away.

**Comment by BrunoScheufler** [comment]
@bensaufley yeah I really have exactly the same problem, but even if I update all the occurrences of the updated version, the "wrapped" component will still be marked with the same error when used via JSX.

**Comment by bensaufley** [comment]
@stevegeek, @andy-ms any input here? With `ts-node`'s [recent decision to default to type-checking](https://github.com/TypeStrong/ts-node/commit/aa9052d6c56967cf225dda1a8bd55cd79183cefc), this breaks by default in tests, forcing me to disable type-checking, which I don't think is a good option.

**Comment by bensaufley** [comment]
UPDATE: I think I've got it @BrunoScheufler

**Comment by BrunoScheufler** [comment]
@bensaufley awesome! The changes look promising 👍

**Comment by bensaufley** [comment]
@BrunoScheufler that PR got merged—give it a try and see if it solves your issue too

**Comment by BrunoScheufler** [comment]
Will do!

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.

**Comment by bensaufley** [comment]
Just for posterity: as far as I can see (it's been a while :grimacing:) this should have been closed with PR #25137 so this is fine to remain closed.
